_studio/mfx_lib/decode/vc1/src/mfx_vc1_decode.cpp

```diff
@@ -1791,6 +1791,24 @@ mfxStatus MFXVideoDECODEVC1::UpdateAllocRequest(mfxVideoParam *par,
     else
         return MFX_ERR_NONE;
 }
+
+void MFXVideoDECODEVC1::SetFrameType(mfxFrameSurface1 &surface) const
+{
+    mfxExtDecodedFrameInfo *extFrameInfo = reinterpret_cast<mfxExtDecodedFrameInfo *>(GetExtendedBuffer(surface.Data.ExtParam, surface.Data.NumExtParam, MFX_EXTBUFF_DECODED_FRAME_INFO));
+    if (extFrameInfo)
```

Comment:
in such situations better to use exit condition instead of huge folded blocks of code:
```
if (!extFrameInfo)
    return;

switch (m_InternMediaDataOut.GetFrameType())
{
...
}
```